### PR TITLE
Improved performance of sw manager topology

### PR DESCRIPTION
**Pull request by niksv**

### CODE REVIEW

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java [hunk 1]
```diff
@@ -242,6 +256,8 @@ private void dispatchWorkerResponse(Message message) throws MessageDispatchExcep
                     message);
             return;
         }
+        //TODO remove
```
Comment (dpoltavets) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java:259]: Should we remove it?

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchValidateService.java [hunk 2]
```diff
@@ -95,6 +92,8 @@ public void dispatchWorkerMessage(InfoData payload, MessageCookie cookie)
             handleMeterEntriesResponse((MeterDumpResponse) payload, cookie);
         } else if (payload instanceof SwitchMeterUnsupported) {
             handleMetersUnsupportedResponse(cookie);
+        } else if (payload instanceof SwitchEntities) {
+            handleExpectedSwitchEntities((SwitchEntities) payload, cookie);
```
Comment (dpoltavets) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchValidateService.java:96]: I would create dispatchHeavyOperationMessage method.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java (hunk 1 above)
Comment (niksv) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java:259]: yes

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchValidateService.java (hunk 2 above)
Comment (niksv) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchValidateService.java:96]: fixed

**Review by dpoltavets — APPROVED** [review]

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java [hunk 3]
```diff
@@ -332,6 +335,16 @@ public void sendCommandToSpeaker(String requestKey, CommandData command, Message
         emit(SpeakerWorkerBolt.INCOME_STREAM, getCurrentTuple(), makeWorkerTuple(requestKey, command, cookie));
     }
 
+    @Override
+    public void runHeavyOperation(String key, SwitchId switchId) {
+        // will never be user, because of carrier decorator
```
Comment (rozdy) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/bolt/SwitchManagerHub.java:340]: user -> used

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchValidateFsm.java [hunk 4]
```diff
@@ -463,14 +459,16 @@ public SwitchId getSwitchId() {
         GROUPS_RECEIVED,
         LOGICAL_PORTS_RECEIVED,
         METERS_UNSUPPORTED,
+        EXPECTED_ENTITIES_BUILD,
```
Comment (rozdy) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchValidateFsm.java:462]: Build -> Built

**Review by rozdy — APPROVED** [review]

**Pull request merged**